ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/ValidatedCommandFactory.java

```diff
@@ -133,6 +138,17 @@ private static Command createForAlterSystemQuery(
               propertyName, propertyValue, propertyName), null);
     }
 
+    // verify that no persistent query is running when attempting to change 'processing.guarantee'
+    final KsqlConfigResolver resolver = new KsqlConfigResolver();
+    final Optional<ConfigItem> resolvedItem = resolver.resolve(propertyName, false);
+    if (resolvedItem.isPresent()
+        && Objects.equals(resolvedItem.get().getPropertyName(), PROCESSING_GUARANTEE_CONFIG)
+        && !context.getPersistentQueries().isEmpty()) {
```

Comment:
do we know this will be true when the command is executed? can there be queries in the cmd topic that have not been started yet or something?

Reply:
cc @rodesai

Reply:
Nope - this should be a sandbox copy of the engine, with the copy being taken during a transaction. So if any other queries slipped in the transaction would fail to commit.